.changeset/good-knives-jam.md

```diff
@@ -0,0 +1,5 @@
+---
+"ts-essentials": patch
+---
+
+Use TypeScript@^4.1.0 because of introduced recursive conditional types
```

Comment:
maybe it makes sense to make it major change as we did when updated typescript peer dependency in [v8.0.0](https://github.com/krzkaczor/ts-essentials/releases/tag/v8.0.0)

Reply:
Yeah, i think it has to be major change as it can break deps.

Reply:
Fixed ✅